src/main/java/com/notably/view/SideBarTreeView.java

```diff
@@ -55,13 +108,23 @@ protected void updateItem(Block block, boolean empty) {
 
             if (empty || block == null) {
                 setText(null);
+                setGraphic(null);
             } else {
                 setText(getNoteTitle());
+                setSelectedProperty(block);
             }
         }
 
         private String getNoteTitle() {
             return getItem().getTitle().getText();
         }
+
+        private void setSelectedProperty(Block block) {
+            if ((block == currentlyOpenedNote.getBlock())) {
```

Comment:
Just to check, this equality check is working fine right? (comparing references is the intention)

Reply:
Yup, it is working fine! thanks @firzanarmani